kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/AbstractWatchManager.java

```diff
@@ -232,9 +232,31 @@ public void close() {
     closeRequest();
     cancelReconnect();
   }
+
+  private WatchEvent contextAwareWatchEventDeserializer(String messageSource) {
+    try {
+      return Serialization.unmarshal(messageSource, WatchEvent.class);
```

Comment:
If the current deserialization mechanism fails we fallback to try to deserialize the more specific class